Firefox Nightly adds `ShadowRoot.setHTML()` and sanitizer parameter

hamishwillee opened the pull request:
FF140 adds remaining sanitizer method/options behind flag in https://bugzilla.mozilla.org/show_bug.cgi?id=1959727

Once FF140 becomes nightly, these would change to "preview".

Changes are:

- `ShadowRoot.setHTML()] - new method
- `options.sanitizer` added to param to `ShadowRoot.setHTMLUnsafe()` and `Document.parseHTMLUnsafe()`

This follows on from #26659

FYI @evilpie I was "somewhat wrong" in #26659. I can add them, just not as a "preview" release. I didn't think of this properly because I was so focused on FF139.

Related docs work being done in https://github.com/mdn/content/issues/38890

evilpie commented:
I think this is still missing the sanitizer option for Element.setHTMLUnsafe.

> Is this planned to ship in 140?

No

hamishwillee commented:
@caugner @evilpie Thanks both
- Added missing method
- Changed to preview - my browser just switched to 140. 
- Changed the subfeature in all methods to experimental.

Should be good now.

caugner commented on api/Element.json:
```diff
@@ -10497,6 +10497,41 @@
+              "chrome_android": "mirror",
+              "edge": "mirror",
+              "firefox": {
+                "version_added": "140"
```
Shouldn't this also be Nightly?
```suggestion
                "version_added": "preview"
```

caugner commented:
> I hadn't caught this, but considering these methods aren't available by default, shouldn't the compat info have the pref instead of a version?

Are these behind a pref even in Nightly?

hamishwillee commented:
@caugner They are behind a flag that is enabled in nightly. 

FYI only, the reason they were previously behind the flag was because when I started this process FF140 was not nightly, and that was the version this went in with.
